### ISSUE TITLE
publish gh release binaries

### ISSUE THREAD

**Issue by softprops** [bazelbuild/bazel-watcher#15]
something like https://github.com/bazelbuild/buildtools/releases would be ideal. using a naming convention what would allow for conditional download based on platform using standard `uname` conventions would be even better

**Comment by achew22** [comment]
Do you have any idea how those are generated? I wonder if I can piggyback on their packaging strategy

**Comment by ittaiz** [comment]
Not sure it's better. I don't object to this but the release cycle of Bazel
can take time which you often don't want to wait for ibazel

**Comment by alexeagle** [comment]
Fair enough, we should keep an independent release process until ibazel is
stable.

**Comment by alexeagle** [comment]
@achew22 how about if we also publish ibazel on NPM? That's the most convenient way to distribute it to Angular users.
/cc @vsavkin

**Comment by achew22** [comment]
100% happy to do that. Do you have a link to docs on how to do that?

**Comment by alexeagle** [comment]
I can set it up and document. Big part is building for each arch on each
release. @mprobst does that for clang-format

**Comment by achew22** [comment]
I sent an email to the Bazel release manager internally so I might try to
piggy back off of their work.

On the plus side, rules_go just added CROSSTOOL support so it shouldn't be
that painful.

**Comment by ittaiz** [comment]
Bazel uses brew and apt-get right?
That is really comfortable (though I imagine more work for the maintainer
than npm)

**Comment by softprops** [comment]
👋 hi, opener of the original issue here. I think the idea of having support for change triggered actions built into bazel would be ideal but for my company that issue becomes a liability in turn around time. We like fast turn arounds when blocked. Thats what ended up facilitating our work on a separate tool https://github.com/meetup/rebazel. I'd worry that having this built into bazels release cycles would not make the turn around process any faster and I'm fact very much slower

**Comment by ittaiz** [comment]
Andrew,
Do you mean to piggy back off their release cycle or their release infra to
have your own releases?

**Comment by achew22** [comment]
@softprops && @ittaiz,

The objective would be to use the CI servers for the heavy lifting of secret storage, scheduling, and uploading. That will make it trivial to actually do releases which I've found makes releases happen more often.

I've put together my requirements/wish list on bazelbuild/continuous-integration#100. Feel free to follow along there. I will keep the conversation as public as I can.

**Comment by Globegitter** [comment]
Now that there is the npm release process, could this not also be leveraged to have the binaries as download on the git releases? It is nice to have it on npm but me and most on my team just prefer getting binaries either directly off of github or through apt/brew but just github releases themselves would be more than great.
